plugins/inputs/icinga2/icinga2.go

```diff
@@ -123,39 +195,79 @@ func (i *Icinga2) Gather(acc telegraf.Accumulator) error {
 		i.client = client
 	}
 
-	requestURL := "%s/v1/objects/%s?attrs=name&attrs=display_name&attrs=state&attrs=check_command"
-
-	// Note: attrs=host_name is only valid for 'services' requests, using check.Attrs.HostName for the host
-	//       'hosts' requests will need to use attrs=name only, using check.Attrs.Name for the host
-	if i.ObjectType == "services" {
-		requestURL += "&attrs=host_name"
+	// For backward config compatibility
+	if i.ObjectType != "" {
+		if len(i.Objects) > 0 {
+			return fmt.Errorf("icinga2.object_type is deprecated, please migrate the config to use only icinga2.objects array only")
+		}
+		i.Objects = []string{i.ObjectType}
 	}
 
-	address := fmt.Sprintf(requestURL, i.Server, i.ObjectType)
+	// Collect /v1/objects
+	for _, objectType := range i.Objects {
+		requestURL := "%s/v1/objects/%s?attrs=name&attrs=display_name&attrs=state&attrs=check_command"
 
-	req, err := http.NewRequest("GET", address, nil)
-	if err != nil {
-		return err
-	}
+		// Note: attrs=host_name is only valid for 'services' requests, using check.Attrs.HostName for the host
+		//       'hosts' requests will need to use attrs=name only, using check.Attrs.Name for the host
+		if objectType == "services" {
+			requestURL += "&attrs=host_name"
+		}
 
-	if i.Username != "" {
-		req.SetBasicAuth(i.Username, i.Password)
-	}
+		address := fmt.Sprintf(requestURL, i.Server, objectType)
 
-	resp, err := i.client.Do(req)
-	if err != nil {
-		return err
-	}
+		resp, err := i.IcingaRequest(address)
+		if err != nil {
+			return err
+		}
 
-	defer resp.Body.Close()
+		result := ResultObject{}
+		err = json.NewDecoder(resp.Body).Decode(&result)
+		if err != nil {
+			return err
+		}
+		err = resp.Body.Close()
+		if err != nil {
+			return err
+		}
 
-	result := Result{}
-	err = json.NewDecoder(resp.Body).Decode(&result)
-	if err != nil {
-		return err
+		i.GatherObjects(acc, result, objectType)
 	}
 
-	i.GatherStatus(acc, result.Results)
+	// Collect /v1/status
+	for _, statusType := range i.Status {
+		requestURL := "%s/v1/status/%s"
+
+		address := fmt.Sprintf(requestURL, i.Server, statusType)
+
+		resp, err := i.IcingaRequest(address)
+		if err != nil {
+			return err
+		}
+
+		tags := map[string]string{
+			"component": statusType,
+		}
+		var fields map[string]interface{}
+
+		switch statusType {
+		case "ApiListener":
+			fields, err = i.ParsePerfdataResponse(resp)
+		case "CIB":
+			fields, err = i.ParseCIBResponse(resp)
+		case "IdoMysqlConnection":
+			fields, err = i.ParsePerfdataResponse(resp)
+		case "IdoPgsqlConnection":
+			fields, err = i.ParsePerfdataResponse(resp)
+		default:
+			return fmt.Errorf("Status endpoint %s is not supported yet", statusType)
```

Comment:
Are there other endpoints?

Reply:
Yes, there are quite a few of them. I would add a link to documentation, but it isn't documented. The number of available endpoints depends on `icinga2` feature configuration:
```
$ icinga2 feature list
Disabled features: command compatlog debuglog elasticsearch gelf graphite icingadb influxdb2 livestatus opentsdb perfdata statusdata syslog
Enabled features: api apiuser checker ido-pgsql influxdb mainlog notification
```
then each feature might expose some statistics.

```
curl -s -k 'https://localhost:5665/v1/status/' | jq .results[].name
"ApiListener"
"CIB"
"CheckResultReader"
"CheckerComponent"
"CompatLogger"
"ElasticsearchWriter"
"ExternalCommandListener"
"FileLogger"
"GelfWriter"
"GraphiteWriter"
"IcingaApplication"
"IdoMysqlConnection"
"IdoPgsqlConnection"
"Influxdb2Writer"
"InfluxdbWriter"
"LivestatusListener"
"NotificationComponent"
"OpenTsdbWriter"
"PerfdataWriter"
"StatusDataWriter"
"SyslogLogger"
```

Reply:
ok sounds like this is fine, but let's change the error message slightly to remove the `yet` part. In the case of a typo, that might not actually be true:

```suggestion
			return fmt.Errorf("Status endpoint %s is not supported", statusType)
```

Reply:
Ok, fixed.